Assets/Scripts/PlayerControls.cs

```diff
@@ -0,0 +1,168 @@
+﻿using UnityEngine;
+using System.Collections;
+
+[RequireComponent( typeof( ActorPhysics ) )]
+public class PlayerControls : MonoBehaviour
+{
+	PlayerActor _actor;
+	ActorPhysics _actorPhysics;
+	bool _jumpButtonDown = false;
+	bool _glideButtonDown = false;
+	bool _isGrabbing = false;
+
+	void Awake()
+	{
+		_actor = GetComponent<PlayerActor>();
+		_actorPhysics = GetComponent<ActorPhysics>();
+
+		SetupStateMethodMap();
+	}
+
+	void Update()
+	{
+		if ( Input.GetKeyDown( KeyCode.Escape ) )
+		{
+			Application.Quit();
+		}
+
+		_isGrabbing = WadeUtils.ValidAxisInput("Grab");
+		_jumpButtonDown = Input.GetButtonDown( "Jump" + WadeUtils.platformName );
+		_glideButtonDown = Input.GetKey( KeyCode.F );
+	}
+
+	void SetupStateMethodMap()
+	{
+		_actorPhysics.RegisterStateMethod( ActorStates.Jumping, Jumping );
+		_actorPhysics.RegisterStateMethod( ActorStates.Falling, Jumping );
+		_actorPhysics.RegisterStateMethod( ActorStates.Grounded, Grounded );
+		_actorPhysics.RegisterStateMethod( ActorStates.Rolling, Rolling );
+		_actorPhysics.RegisterStateMethod( ActorStates.Climbing, Climbing );
+	}
+
+	void Jumping()
+	{
+		if ( _jumpButtonDown )
+		{
+			_actorPhysics.JumpCheck();
+		}
+
+		if ( _isGrabbing )
+		{
+			_actorPhysics.ClimbCheck();
+		}
+
+		_actorPhysics.RollCheck();
+
+		if ( _glideButtonDown )
+		{
+			_actorPhysics.GlideMovement( GetMoveDirection() );
+		}
+		else
+		{
+			_actorPhysics.JumpMovement( GetMoveDirection() );
+		}
+	}
+
+	void Rolling()
+	{
+		_actorPhysics.RollCheck();
+		_actorPhysics.RollMovement();
+	}
+
+	void Climbing()
+	{
+		if ( _actorPhysics.ClimbCheck() &&
+			_isGrabbing &&
+			_actor.actorStats.CanUseStat( Stat.Stamina ) )
+		{
+			_actor.actorStats.StartUsingStat( Stat.Stamina );
+			_actorPhysics.ClimbSurface( GetInput() );
+		}
+		else
+		{
+			_actor.actorStats.StopUsingStat( Stat.Stamina );
+			_actorPhysics.StopClimbing();
+		}
+	}
+
+	void Grounded()
+	{
+		if ( _jumpButtonDown )
+		{
+			_actorPhysics.JumpCheck();
+		}
+
+		if ( _isGrabbing )
+		{
+			_actorPhysics.ClimbCheck();
+		}
+
+		_actorPhysics.RollCheck();
+
+		GroundMovement();
+	}
+
+	void Gliding()
+	{
+		if ( _jumpButtonDown )
```

Comment:
Nothing, unless you're *reaaaaaalllllly* close to the ground. This is to match the behavior when jumping, which let's you jump again when you're just about on the ground but haven't quite landed yet.